src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/Models/PerformanceMetricData.cs

```diff
@@ -12,28 +13,26 @@ public class PerformanceMetricData
         public double previousTickTime;
         public int framesSincePreviousTick;
         public int fps;
-        public long? cpuTimeSincePreviousTick;
-        public long? memory;
-        public long? gcMemory;
-        public EngineStatsData engineStats;
+        public List<PerFrameStatisticsData> perFrameStatistics;
 
         public void WriteToStringBuilder(StringBuilder stringBuilder)
         {
-            stringBuilder.Append("{\"previousTickTime\":");
+            stringBuilder.Append("{\n\"previousTickTime\":");
             DoubleJsonConverter.WriteToStringBuilder(stringBuilder, previousTickTime);
-            stringBuilder.Append(",\"framesSincePreviousTick\":");
+            stringBuilder.Append(",\n\"framesSincePreviousTick\":");
             IntJsonConverter.WriteToStringBuilder(stringBuilder, framesSincePreviousTick);
-            stringBuilder.Append(",\"fps\":");
+            stringBuilder.Append(",\n\"fps\":");
             IntJsonConverter.WriteToStringBuilder(stringBuilder, fps);
-            stringBuilder.Append(",\"cpuTimeSincePreviousTick\":");
-            LongJsonConverter.WriteToStringBuilderNullable(stringBuilder, cpuTimeSincePreviousTick);
-            stringBuilder.Append(",\"memory\":");
-            LongJsonConverter.WriteToStringBuilderNullable(stringBuilder, memory);
-            stringBuilder.Append(",\"gcMemory\":");
-            LongJsonConverter.WriteToStringBuilderNullable(stringBuilder, gcMemory);
-            stringBuilder.Append(",\"engineStats\":");
-            engineStats.WriteToStringBuilder(stringBuilder);
-            stringBuilder.Append("}");
+            stringBuilder.Append(",\n\"perFrameStatistics\":[\n");
+            for (int i = 0, n = perFrameStatistics.Count; i < n; ++i)
```

Comment:
Functionally this is fine, but it is inconsistent stylistically with all the other code

Reply:
I've updated the code to match the style of the others,  let me know if there's anything I missed!